### PR TITLE
Adopt Laminas CI Workflow

### PR DESCRIPTION
**Pull request by Slamdunk**
Blocks https://github.com/laminas/laminas-server/pull/25

### CODE REVIEW

**Comment by froschdesign** [comment]
@Slamdunk 
The `composer.lock` file must be added.

**Comment by Slamdunk** [comment]
I have never understood how and why: I would generate it under my local PC settings, which is arbitrary and way different from CI.

Is this ok? If so, how is it useful in the first place?

**Comment by froschdesign** [comment]
> Is this ok? If so, how is it useful in the first place?

See the checks running on this pull request, the CI workflow is still missing.

See also:

* https://github.com/laminas/laminas-ci-matrix-action#create-a-ci-matrix-for-use-in-a-github-action
* https://github.com/laminas/laminas-feed/pull/35

**Comment by Slamdunk** [comment]
I've committed and pushed it. Still, I feel you didn't answer my question:

1. https://github.com/laminas/laminas-ci-matrix-action#create-a-ci-matrix-for-use-in-a-github-action doesn't require a `composer.lock` committed:
    > Whether to run against a "locked" set of dependencies based on the presence of a composer.lock file.
2. https://github.com/laminas/laminas-feed/pull/35 doesn't say *how* someone should generate it

**Comment by Slamdunk** [comment]
Now build passes, PSalm fails due to https://github.com/laminas/laminas-continuous-integration-action/issues/9

**Comment by froschdesign** [comment]
> Still, I feel you didn't answered my question…

You are absolutely right, but I am not the expert on this topic and so I can not much help here, unfortunately only with some hints.

**Comment by Slamdunk** [comment]
Do you think we can close [2.10.0](https://github.com/laminas/laminas-server/milestone/4) milestone today?

**Review by froschdesign — APPROVED** [review]

**Comment by Slamdunk** [comment]
I see you put this into `2.10.0` milestone, but there's no `2.10.x` branch

**Comment by froschdesign** [comment]
> I see you put this into 2.10.0 milestone, but there's no 2.10.x branch

Right, I just saw that too.
I'm on the road and need to check what I can do via the GitHub app.

**Comment by Slamdunk** [comment]
>  I'm on the road and need to check what I can do via the GitHub app.

Oh, never mind, please wait for the proper time and location to work on this :wink:

**Comment by weierophinney** [comment]
> 've committed and pushed it. Still, I feel you didn't answer my question:
> 
> 1. https://github.com/laminas/laminas-ci-matrix-action#create-a-ci-matrix-for-use-in-a-github-action doesn't require a `composer.lock` committed:
>    > Whether to run against a "locked" set of dependencies based on the presence of a composer.lock file.

It's needed for non-unit test checks. The reason is because updates to QA tooling can quickly lead to different results on different runs of the same command. As examples:

- Things not previously flagged as CS issues now are.
- Things not previously caught by static analysis now are, even though no relevant code changes were made.

As such, we want to commit a lockfile so that these remain idempotent between invocations on the same set of code.

> 2. [laminas/laminas-feed#35](https://github.com/laminas/laminas-feed/pull/35) doesn't say _how_ someone should generate it

You're right, so I wrote up the process here: https://gist.github.com/weierophinney/b003e50c3c2667d08076caf31ebd36a4 (and just added the bit about Psalm cache directory issues as well).

**Comment by weierophinney** [comment]
@Slamdunk Okay, the laminas-continuous-integration-action is updated to resolve the Psalm caching issue. Now the check is failing due to actual errors!

Ping me when resolved, and I'll get this merged and released.

File: psalm-baseline.xml [hunk 1]
```diff
@@ -176,11 +160,16 @@
     <PropertyNotSetInConstructor occurrences="1">
       <code>$name</code>
     </PropertyNotSetInConstructor>
+    <RedundantCastGivenDocblockType occurrences="4">
+      <code>(bool) $flag</code>
+      <code>(string) $description</code>
+      <code>(string) $name</code>
+      <code>(string) $type</code>
+    </RedundantCastGivenDocblockType>
```
Comment (Slamdunk) [inline psalm-baseline.xml:168]: These redundant casts cannot be purged as they would result in a BC break, see:

https://github.com/laminas/laminas-server/blob/6488e14a50c7308c2d1aecc19e8305b27e560da2/test/Method/ParameterTest.php#L70-L78

**Comment by Slamdunk** [comment]
Ping @weierophinney

File: psalm-baseline.xml (hunk 1 above)
Comment (weierophinney) [reply psalm-baseline.xml:168]: You can suppress them individually, or add them to the baseline, then.

**Pull request merged**